lib/model.js

```diff
@@ -106,45 +104,54 @@ module.exports = class Model extends EventEmitter {
           sendImmediately: true
         };
       }
-    }).then(() => {
+    })
+    .then(() => {
       const startTime = process.hrtime();
       let timeoutTimer;
 
-      const _callback = (err, data, statusCode) => {
-        if (timeoutTimer) {
-          clearTimeout(timeoutTimer);
-          timeoutTimer = null;
-        }
-
-        const endTime = process.hrtime();
-        const responseTime = timeDiff(startTime, endTime);
-
-        if (err) {
-          this.emit('fail', err, data, originalSettings, statusCode, responseTime);
-        } else {
-          this.emit('success', data, originalSettings, statusCode, responseTime);
-        }
-        if (typeof callback === 'function') {
-          callback(err, data, responseTime);
-        }
-      };
-
-      this._request(settings, (err, response) => {
-
-        if (err) {
-          if (err.code === 'ETIMEDOUT') {
-            err.message = 'Connection timed out';
-            err.status = 504;
+      return new Promise((resolve, reject) => {
+
+        const _callback = (err, data, statusCode) => {
+          if (timeoutTimer) {
+            clearTimeout(timeoutTimer);
+            timeoutTimer = null;
+          }
+
+          const endTime = process.hrtime();
+          const responseTime = timeDiff(startTime, endTime);
+
+          if (err) {
+            this.emit('fail', err, data, originalSettings, statusCode, responseTime);
+          } else {
+            this.emit('success', data, originalSettings, statusCode, responseTime);
+          }
+          if (err) {
+            reject(err);
+          } else {
+            resolve(data);
           }
-          err.status = err.status || (response && response.statusCode) || 503;
-          return _callback(err, null, err.status);
-        }
-        return this.handleResponse(response, _callback);
+        };
+
+        this._request(settings, (err, response) => {
+
+          if (err) {
+            if (err.code === 'ETIMEDOUT') {
+              err.message = 'Connection timed out';
+              err.status = 504;
+            }
+            err.status = err.status || (response && response.statusCode) || 503;
+            return _callback(err, null, err.status);
+          }
+          return this.handleResponse(response, _callback);
+        });
       });
 
     });
 
-    this.emit('sync', originalSettings);
+    if (typeof callback === 'function') {
+      return promise.then((data) => callback(null, data), callback);
+    }
+    return promise;
```

Comment:
If I were being clever then I'd have just extended the class, with a cheeky `Bluebird.promisify` around the methods. I am not clever.